situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/processor/oteltrace/RawTrace.java

```diff
@@ -0,0 +1,160 @@
+            checkNotNull(traceId, "traceId cannot be null");
+            this.traceId = traceId;
+            return this;
+        }
+
+        public RawTraceBuilder setSpanId(final String spanId) {
+            checkNotNull(spanId, "spanId cannot be null");
+            this.spanId = spanId;
+            return this;
+        }
+
+        public RawTraceBuilder setParentSpanId(final String parentSpanId) {
+            this.parentSpanId = parentSpanId;
+            return this;
+        }
+
+        public RawTraceBuilder setName(final String name) {
+            this.name = name;
+            return this;
+        }
+
+        public RawTraceBuilder setStartTime(final String startTime) {
+            this.startTime = startTime;
+            return this;
+        }
+
+        public RawTraceBuilder setEndTime(final String endTime) {
+            this.endTime = endTime;
+            return this;
+        }
+
+        public RawTraceBuilder setInstrumentationName(final String instrumentationName) {
+            this.instrumentationName = instrumentationName;
+            return this;
+        }
+
+        public RawTraceBuilder setInstrumentationVersion(final String instrumentationVersion) {
+            this.instrumentationVersion = instrumentationVersion;
+            return this;
+        }
+
+        public RawTraceBuilder setServiceName(final String serviceName) {
+            this.serviceName = serviceName;
+            return this;
+        }
+
+        public RawTraceBuilder setResourceAttributes(final Map<String, String> resourceAttributes) {
+            this.resourceAttributes = resourceAttributes;
+            return this;
+        }
+
+        public RawTrace build() {
+            return new RawTrace(this);
+        }
+
+    }
+
+    private static String convertStringNanosToISO8601(final String stringNanos) {
+        final BigDecimal nanos = new BigDecimal(stringNanos);
+        final long epochSeconds = nanos.divide(MILLIS_TO_NANOS.multiply(SEC_TO_MILLIS), RoundingMode.DOWN).longValue();
+        final int nanoAdj = nanos.remainder(MILLIS_TO_NANOS.multiply(SEC_TO_MILLIS)).intValue();
+        return Instant.ofEpochSecond(epochSeconds, nanoAdj).toString();
+    }
+
+
+    public static RawTrace buildFromProto(Resource resource, Span spans, InstrumentationLibrary instrumentationLibrary) {
+        final String SERVICE_NAME = "service.name";
+
+        Map<Object, Object> map = resource.getAttributesList().stream().collect(Collectors.toMap(KeyValue::getKey, KeyValue::getValue));
+        final Map<String, String> resourceAttributesMap = new HashMap<>();
+        for (Map.Entry<Object, Object> entry : map.entrySet()) {
+            if (entry.getValue() instanceof String) {
+                resourceAttributesMap.put((String) entry.getKey(), (String) entry.getValue());
+            }
+            else {
+                resourceAttributesMap.put((String) entry.getKey(), String.valueOf(entry.getValue()));
```

Comment:
We shouldn't do string.valueOf(), because it could be other [types](https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/common/v1/common.proto#L30-L36).